src/AutomationTests/SnapshotCommandUnitTests.cs

```diff
@@ -1,404 +1,327 @@
-                ShimSelectAction selectAction = new ShimSelectAction()
-                {
-                    SetCandidateElementA11yElement = (element) => { },
-                    Select = () => true,
-                    POIElementContextGet = () => new ElementContext(
-                        new A11yElement
-                        {
-                            ScanResults = new ScanResults(),
-                            Children = new List<A11yElement>
-                            {
-                                new A11yElement
-                                {
-                                    ScanResults = new ScanResults(),
-                                    Children = new List<A11yElement>(),
-                                }
-                            }
-                        }),
-                };
-
-                InitializeShims(shimTargetElementLocator: true,
-                    selectAction: selectAction, elementBoundExceeded: false, shimUiFramework: true,
-                    setTestModeSucceeds: true, shimScreenCapture: true, shimSnapshot: true, shimSarif: true);
-
-                var config = Config.Builder.ForProcessId(-1).Build();
-                var resultsAssembler = new ScanResultsAssembler();
-
-                var results = SnapshotCommand.Execute(config, OutputFileHelperStub, resultsAssembler);
-
-                Assert.AreEqual(0, results.ErrorCount);
-            }
+            _scanToolsMock.Setup(x => x.TargetElementLocator).Returns(_targetElementLocatorMock.Object);
+            _scanToolsMock.Setup(x => x.InternalScanner).Returns(_internalScannerMock.Object);
+            _scanToolsMock.Setup(x => x.ResultsAssembler).Returns(_resultsAssemblerMock.Object);
+
+            _targetElementLocatorMock.Setup(x => x.LocateRootElement(It.IsAny<int>())).Returns(new A11yElement());
+
+            var expectedResults = new ScanResults();
+            expectedResults.ErrorCount = 0;
+            InitResultsCallback(expectedResults);
+
+            // In addition to throwing an ArgumentNullException
+            // The following call would cause mock exceptions for IInternalScann.CaptureScreenshot and IInternalScanner.SaveA11yTestFile.
+            // 
+            var config = Config.Builder
+                .ForProcessId(-1)
+                .WithOutputFileFormat(OutputFileFormat.A11yTest)
+                .Build();
+
+            var actualResults = SnapshotCommand.Execute(config, _scanToolsMock.Object);
+            Assert.IsNull(actualResults.OutputFile.A11yTest);
+
+            _scanToolsMock.VerifyAll();
+            _targetElementLocatorMock.VerifyAll();
+            _internalScannerMock.VerifyAll();
+            _resultsAssemblerMock.VerifyAll();
         }
 
         [TestMethod]
-        [Timeout(2000)]
-        public void Execute_OnlyFailuresInPOI_SavesFile_ReturnsComplete_FailuresOnly()
+        [Timeout(1000)]
+        public void Execute_OutputFileHelperIsNull_ThrowsException()
```

Comment:
Your other null checking tests are phrased Null<thing>_ThrowsException.
```suggestion
        public void Execute_NullOutputFileHelper_ThrowsException()
```